src/modules/gov/proposal.rs

```diff
@@ -0,0 +1,113 @@
+//! Definition of domain `Proposal`.
+
+use ibc_proto::cosmos::gov::v1beta1::Proposal as RawProposal;
+use ibc_proto::cosmos::gov::v1beta1::ProposalStatus;
+use ibc_proto::cosmos::gov::v1beta1::TallyResult;
+use ibc_proto::google::protobuf::{Any, Timestamp};
+use ibc_proto::protobuf::Protobuf;
+
+use super::error::Error;
+use crate::modules::bank::util::Coin;
+
+pub(crate) const TYPE_URL: &str = "/cosmos.gov.v1beta1.Proposal";
+
+#[derive(Clone, Debug)]
+pub struct Proposal {
```

Comment:
What is the difference between this type and `UpgradeProposal` in ibc-rs? Shouldn't this go in ibc-rs as well?

Reply:
- The main purpose of `Proposal` is to track the status of a proposal we get from `MsgSubmitProposal`, particularly when going through the voting process. Regardless of what the content is. Once the proposal is passed, the app will call the upgrade module to apply the upgrade.
 - While the `UpgradeProposal` is one of the possible **content** types (extracted from `Any`), which contains an `upgraded_client_state` and because of this it’s related to ics02_client and placed as a helper type within IBC-rs, unlike `Proposal, which is for the Gov module.

- I included `Proposal`, so be able to query the plan_height through the `gaiad` cli (as a shortcut/temporary way) needed for the CI. Though, neither could query the upgrade `Plan` nor the `Proposal` through either Gov or Upgrade module.
Apparently I came across the same issue discussed [here](https://github.com/informalsystems/basecoin-rs/issues/89#issuecomment-1480923330) and `gaiad` not working properly with CometBFT v0.37. 
Sounds like we should add a separate **basecoin-cli** crate to enable us to query the store for different datagrams, at least needed for integration tests. I can complete that part after adding that CLI. WDYT?
Anyways, you can also try the upgrade locally, it works.